src/Client.ts

```diff
@@ -0,0 +1,236 @@
+      Logger.error('Could not construct Client class: missing bot token in envvars', {
+        eventType: 'initError',
+        error: 'missing bot token in envvars',
+      });
+      throw new Error('Could not construct Client class: missing bot token in envvars');
+    }
+    if (!process.env.BOT_PREFIX) {
+      Logger.error('Could not construct Client class: missing bot prefix in envvars', {
+        eventType: 'initError',
+        error: 'missing bot prefix in envvars',
+      });
+      throw new Error('Could not construct Client class: missing bot prefix in envvars');
+    }
+    if (!process.env.CLIENT_ID) {
+      Logger.error('Could not construct Client class: missing app client ID in envvars', {
+        eventType: 'initError',
+        error: 'missing app client ID in envvars',
+      });
+      throw new Error('Could not construct Client class: missing app client ID in envvars');
+    }
+    this.settings.clientID = process.env.CLIENT_ID;
+    this.settings.token = process.env.BOT_TOKEN;
+    this.settings.prefix = process.env.BOT_PREFIX;
+    this.settings.apiKeys.catAPI = process.env.CAT_API_KEY;
+    this.settings.apiKeys.unsplash = process.env.UNSPLASH_ACCESS_KEY;
+
+    if (!process.env.ACMURL_USERNAME) {
+      Logger.error('Could not construct Client class: missing ACMURL username in envvars', {
+        eventType: 'initError',
+        error: 'missing ACMURL username in envvars',
+      });
+      throw new Error('Could not construct Client class: missing ACMURL username in envvars');
+    }
+    if (!process.env.ACMURL_PASSWORD) {
+      Logger.error('Could not construct Client class: missing ACMURL password in envvars', {
+        eventType: 'initError',
+        error: 'missing ACMURL password in envvars',
+      });
+      throw new Error('Could not construct Client class: missing ACMURL password in envvars');
+    }
+    if (!process.env.MEMBERSHIP_PORTAL_API_USERNAME) {
+      Logger.error('Could not construct Client class: missing Membership Portal API username in envvars', {
+        eventType: 'initError',
+        error: 'missing Membership Portal API username in envvars',
+      });
+      throw new Error('Could not construct Client class: missing Membership Portal API username in envvars');
+    }
+    if (!process.env.MEMBERSHIP_PORTAL_API_PASSWORD) {
+      Logger.error('Could not construct Client class: missing Membership Portal API password in envvars', {
+        eventType: 'initError',
+        error: 'missing Membership Portal API password in envvars',
+      });
+      throw new Error('Could not construct Client class: missing Membership Portal API password in envvars');
+    }
+    this.settings.acmurl.username = process.env.ACMURL_USERNAME;
+    this.settings.acmurl.password = process.env.ACMURL_PASSWORD;
+    this.settings.portalAPI.username = process.env.MEMBERSHIP_PORTAL_API_USERNAME;
+    this.settings.portalAPI.password = process.env.MEMBERSHIP_PORTAL_API_PASSWORD;
```

Comment:
These are all from BreadBot, they can be removed; we don't use these here. If we need them later, we will put them back in, but I don't think we will need them, at least for a while.